File: Example/cenarius/src/main/java/com/m/cenarius/Web/Interceptor/WebViewActivity.java
package com.m.cenarius.Web.Interceptor;

import android.app.Activity;
import android.os.Bundle;
import android.webkit.WebResourceRequest;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import com.alibaba.fastjson.JSONObject;
import com.m.cenarius.R;
import com.m.cenarius.R2;
import com.m.cenarius.Route.Route;
import com.m.cenarius.Update.UpdateManager;

import java.io.File;

import butterknife.BindView;
import butterknife.ButterKnife;

public class WebViewActivity extends Activity {

    @BindView(R2.id.webView)
    WebView webView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_web_view);

        ButterKnife.bind(this);

        setupWebView();

        webView.setWebViewClient(new WebViewClient() {
            @Override
            public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
                String url = request.getUrl().toString();
                if (!Interceptor.perform(url, webView.getContext())) {
                    view.loadUrl(url);
                }
                return super.shouldOverrideUrlLoading(webView, request);
            }
        });

        JSONObject params = Route.getParamsJsonObject(this);
        if (params != null) {
            String file = params.getString("file");
            String url = params.getString("url");
            if (file != null) {
                File urlFile = UpdateManager.getCacheUrl(file);
                if (urlFile.exists()) {
                    url = "file://" + urlFile.getAbsolutePath();
                    webView.loadUrl(url);
                }
            } else if (url != null) {
                webView.loadUrl(url);
            }
        }
    }

    private void setupWebView() {
        WebSettings webSettings = webView.getSettings();
        webSettings.setAllowContentAccess(true);
        webSettings.setAllowFileAccess(true);
        webSettings.setAllowFileAccessFromFileURLs(true);
        webSettings.setAllowUniversalAccessFromFileURLs(true);
        webSettings.setAppCacheEnabled(false);
        webSettings.setBuiltInZoomControls(true);
        webSettings.setDisplayZoomControls(false);
        webSettings.setJavaScriptEnabled(true);
        webSettings.setGeolocationEnabled(true);
        webSettings.setLoadWithOverviewMode(true);
        webSettings.setUseWideViewPort(true);
        webSettings.setMixedContentMode(WebSettings.MIXED_CONTENT_ALWAYS_ALLOW);
        webSettings.setLayoutAlgorithm(WebSettings.LayoutAlgorithm.NORMAL);
    }


}

File: cenarius/src/main/java/com/m/cenarius/Web/Interceptor/WebViewActivity.kt
package com.m.cenarius.Web.Interceptor

import android.app.Activity
import android.os.Bundle
import android.webkit.WebSettings
import android.webkit.WebView
import android.webkit.WebViewClient

import com.m.cenarius.Extension.getParamsJsonObject
import com.m.cenarius.R

import kotlinx.android.synthetic.main.activity_web_view.*

class WebViewActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_web_view)

        setupWebView()

        webView.setWebViewClient(object : WebViewClient() {
            override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
                return Interceptor.perform(url, webView.context)
            }
        })

        val params = getParamsJsonObject()
        if (params != null) {
            val url = params.getString("url")
            webView.loadUrl(url)
        }
    }

    private fun setupWebView() {
        val webSettings = webView.settings
        webSettings.allowContentAccess = true
        webSettings.allowFileAccess = true
        webSettings.allowFileAccessFromFileURLs = true
        webSettings.allowUniversalAccessFromFileURLs = true
        webSettings.setAppCacheEnabled(false)
        webSettings.builtInZoomControls = true
        webSettings.displayZoomControls = false
        webSettings.javaScriptEnabled = true
        webSettings.setGeolocationEnabled(true)
        webSettings.loadWithOverviewMode = true
        webSettings.useWideViewPort = true
        webSettings.mixedContentMode = WebSettings.MIXED_CONTENT_ALWAYS_ALLOW
        webSettings.layoutAlgorithm = WebSettings.LayoutAlgorithm.NORMAL
    }

    override fun onBackPressed() {
        if (webView.canGoBack()) {
            webView.goBack()
        } else {
            super.onBackPressed()
        }
    }
}
